src/mame/mr/nl_dribling.cpp

```diff
@@ -0,0 +1,524 @@
+	ALIAS(4, IC.11)
+	ALIAS(5, IC.1)
+	ALIAS(GND, IC.11)
+	NET_C(GND, IC.5, IC.6, IC.10, IC.9, IC.12, IC.13)
+}
+#else
+static NETLIST_START(LM339) 
+{
+	// * CONNECTIONS:   NON-INVERTING INPUT
+	// *                | INVERTING INPUT
+	// *                | | POSITIVE POWER SUPPLY
+	// *                | | | NEGATIVE POWER SUPPLY
+	// *                | | | | OPEN COLLECTOR OUTPUT
+	// *                | | | | |
+	// *                1 2 3 4 5
+	NET_MODEL("LM339A_TI_QIN PNP(IS=800.0E-18 BF=2.000E3)")
+	NET_MODEL("LM339A_TI_QMI NPN(IS=800.0E-18 BF=1002)")
+	NET_MODEL("LM339A_TI_QMO NPN(IS=800.0E-18 BF=1000 CJC=1E-15 TR=807.4E-9)")
+	NET_MODEL("LM339A_TI_QOC NPN(IS=800.0E-18 BF=20.29E3 CJC=1E-15 TF=942.6E-12 TR=543.8E-9)")
+	NET_MODEL("LM339A_TI_DX D(IS=800.0E-18)")
+	ALIAS(1, VI1.2)
+	ALIAS(2, VI2.2)
+	ALIAS(3, F10.ON)
+	ALIAS(4, Q3.E)
+	ALIAS(5, Q5.C)
+	DIODE(DP, "LM339A_TI_DX")
+	VCVS(E1, 1)
+	CCCS(F10, 1)
+	CS(IEE, 0.0001)
+	QBJT_EB(Q1, "LM339A_TI_QIN")
+	QBJT_EB(Q2, "LM339A_TI_QIN")
+	QBJT_EB(Q3, "LM339A_TI_QMO")
+	QBJT_EB(Q4, "LM339A_TI_QMI")
+	QBJT_EB(Q5, "LM339A_TI_QOC")
+	RES(RP, 46.300000)
+	VS(V1, 0)
+	VS(VI1, 0.75)
+	VS(VI2, 0.75)
+	NET_C(E1.OP, F10.IP)
+	NET_C(Q2.C, Q3.B, Q4.C, Q4.B)
+	NET_C(VI1.1, Q1.B)
+	NET_C(VI2.1, Q2.B)
+	NET_C(IEE.2, Q1.E, Q2.E)
+	NET_C(F10.IN, V1.1)
+	NET_C(F10.ON, IEE.1, DP.K, RP.1)
+	NET_C(F10.OP, Q1.C, Q3.C, E1.IP)
+	NET_C(V1.2, Q5.B)
+	NET_C(Q3.E, Q4.E, E1.ON, E1.IN, Q5.E, DP.A, RP.2)
+}
+#endif
+
+
+
+// CALCIO_A and CALCIO_B differ only for the parameters of 3 capacitors.
+static NETLIST_START(calcio)
+{
+	DEFPARAM(CAP1, 33)
+	DEFPARAM(CAP2, 68)
+	ANALOG_INPUT(I_V5, 5)
+	ANALOG_INPUT(I_V6, 6)
+	ANALOG_INPUT(I_V12, 12)
+	QBJT_EB(Q8, "BC239C")
+	QBJT_EB(Q9, "BC239C")
+	ALIAS(INPUT, R45.2)
+	RES(R45, 420) 
+	RES(R46, RES_K(8.2))  
+	RES(R48, RES_K(220)) 
+	RES(R49, RES_K(56)) 
+	RES(R50, RES_K(56)) 
+	RES(R51, RES_M(1)) 
+	RES(R52, RES_K(68)) 
+	RES(R53, RES_K(82))  
+	RES(R54, RES_K(1.9))  // WAS POT
+	RES(R55, RES_K(10))
+	RES(R56, RES_K(100))
+	RES(R57, RES_K(10))
+	RES(R58, RES_K(3.9))
+	CAP(C29, CAP_U(4.2))
+	CAP(C30, CAP_N($(@.CAP2)))	
+	CAP(C31, CAP_N(150)) 
+	CAP(C32, CAP_U(4.7))
+	CAP(C33, CAP_N($(@.CAP1))) 
+	CAP(C34, CAP_N($(@.CAP1))) 
+	NET_C(INPUT, R46.1)
+	ALIAS(GND, R45.1)
+	NET_C(R46.2, C33.2, C34.1)
+	NET_C(C33.1, R49.1, C29.1)
+	NET_C(C34.2, R50.2, R51.1, Q8.B)
+	NET_C(R49.2, C30.2, R50.1)
+	NET_C(C30.1, GND)
+	NET_C(C29.2, R52.1, R51.2, R53.1, Q8.C)
+	NET_C(R52.2, I_V5)
+	NET_C(Q8.E, R54.2)
+	NET_C(R54.1, GND)
+	NET_C(R53.2, C31.1)
+	NET_C(C31.2, R48.1, R56.2, Q9.B)
+	NET_C(R48.2, I_V5)
+	NET_C(R56.1, GND)
+	NET_C(Q9.C, R55.1)
+	ALIAS(OUTPUT, Q9.C)
+	NET_C(R55.2, I_V5)
+	NET_C(Q9.E, R57.2, C32.2)
+	NET_C(R57.1, GND)
+	NET_C(C32.1, R58.2)
+	NET_C(R58.1, GND)
+}
+
+// STOP_PALLA AND PARATA are very similar. PARATA requires JFETs and 
+// MOSFET EMULATION doesn't work. So using the same circuit with different
+// values to emulate the sound.
+static NETLIST_START(stop_palla_parata) 
+{
```

Comment:
What are the JFETs being used for?  What kind of MOSFETs did you use to try simulating them?

Reply:
That's a good question and the circuit simulator seems to show that the effect is to clip below 2.5V to output, meaning let only the portion above it pass through.
I've used the suggestion in the netlist FAQ used all over JFETs are required but it would not produce results (a bunch of NaN).
The current solution though comes pretty close given the schematics being identical except for the output.

Reply:
OK, I looked at the schematic.  It definitely isn’t “identical except for the output stage.”  The first JFET is part of the function generator.

The first JFET has its source and gate tied together on the LM339 output:
* When the LM339 is pulling its output low, the JFET will operate in constant current mode.  The timing capacitor will discharge via the JFET and the LM339 output.  The discharge curve will be a linear ramp.
* When the LM339 is not pulling its output low, the gate-drain region of the JFET will be forward-biased, so it will present a low impedance in series with the 2.2kΩ resistor.  The capacitor will charge as though it’s effectively connected to the 2.2kΩ resistor directly.  The charge curve will be a normal RC charge curve.

The second JFET is connected in source follower configuration.  It presents a high-impedance input and low-impedance output.  This prevents the mixing stage from interfering with the function generator.

The circuit is supposed to generate a kind of sawtooth waveform, rising with a relatively quick RC charge curve, and falling with a linear ramp.  You’ll need to use *something* to substitute for the JFETs.

Once again, this won’t work with an op-amp.  With an op-amp, the 2.2kΩ resistor will be effectively shorted when the op-amp is driving its output high.  This will mean the capacitor will charge far quicker than it should.

Please answer this question:

>  What kind of MOSFETs did you use to try simulating them?

Did you substitute depletion mode MOSFETs with appropriate Vgs=0 current?  Did you try using a depletion mode MOSFET with a diode connected across the drain and source for the first JFET?